src/services/ShopwareDataHelpers.ts

```diff
@@ -254,4 +255,28 @@ export function extractIdFromUrl(url: string): string | null {
 type OrderStatus = 'cancel' | 'complete' | 'reopen' | 'process';
 export const setOrderStatus = async (orderId: string, orderStatus: OrderStatus, adminApiContext: AdminApiContext): Promise<APIResponse> => {
     return await adminApiContext.post(`./_action/order/${orderId}/state/${orderStatus}`);
+};
+
+export const getPromotionWithDiscounts = async (promotionId: string, adminApiContext: AdminApiContext): Promise<Promotion[]> => {
+    const resp = await adminApiContext.post('search/promotion', {
+        data: {
+            limit: 1,
+            associations: {
+              discounts: {
+                  limit: 10,
+                  type: 'equals',
+                  field: 'promotionId',
+                  value: promotionId,
+              },
+            },
+            filter: [{
+                type: 'equals',
+                field: 'id',
+                value: promotionId,
+            }],
+        },
+    });
+
+    const { data: promotion } = (await resp.json()) as { data: Promotion[] };
+    return promotion;
```

Comment:
As this method is explicitly searching for a specific promotion, it should not return an array but a single promotion.

Reply:
added that the method returns just one promotion.